File: src/simulation/culture/interactionmodel/InteractionModel.java
package simulation.culture.interactionmodel;

import simulation.World;
import simulation.culture.Event;

import java.util.Collection;

/**
 * Represents general model by which World changes.
 */
public interface InteractionModel {
    void turn(World world);

    void geologicTurn(World world);

    Collection<Event> getEvents();

    void clearEvents();
}

File: src/shmp/simulation/interactionmodel/InteractionModel.kt
package shmp.simulation.interactionmodel

import shmp.simulation.World
import shmp.simulation.event.EventLog

/**
 * Represents general model by which World changes.
 */
interface InteractionModel {
    fun turn(world: World)
    fun geologicTurn(world: World)
    val eventLog: EventLog
}